File: front/src/app/models/Usuario.ts
import { Curso } from './Curso';
import { Plan } from './Plan';

export class Usuario {
	public id_usuario: number;

	public nombre_usuario: string;

	public foto_usuario: string[];

	public presentacion: string;

	public roll_usuario?: number;

	public plan_usuario?: Plan;

	public cursos_usuario?: Curso[];

	public fecha_registro_usuario?: Date;

	constructor(_id_usuario: number, _nombre_usuario: string, _foto_usuario: string[], _presentacion: string, _roll_usuario?: number, _plan_usuario?: Plan, _cursos_usuario?: Curso[], _fecha_registro_usuario?: Date) {
		this.id_usuario = _id_usuario;
		this.nombre_usuario = _nombre_usuario;
		this.foto_usuario = _foto_usuario;
		this.presentacion = _presentacion;
		this.roll_usuario = _roll_usuario;
		this.plan_usuario = _plan_usuario;
		this.cursos_usuario = _cursos_usuario;
		this.fecha_registro_usuario = _fecha_registro_usuario;
	}
}

File: back_base/src/main/java/com/sovereingschool/back_base/Models/Usuario.java
package com.sovereingschool.back_base.Models;

import java.io.Serializable;
import java.util.Date;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.JoinTable;
import jakarta.persistence.ManyToMany;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import jakarta.persistence.Temporal;
import jakarta.persistence.TemporalType;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@ToString
@Builder
@Entity
@Table(name = "usuario")
public class Usuario implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id_usuario;

    @Column(nullable = false)
    private String nombre_usuario;

    @Column(unique = true, columnDefinition = "text[]")
    private List<String> foto_usuario;

    @Column(length = 1500)
    private String presentacion;

    @Column(nullable = false, name = "roll_usuario")
    @Enumerated(EnumType.STRING) // Esto es importante
    private RoleEnum roll_usuario;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "plan_usuario", referencedColumnName = "id_plan")
    private Plan plan_usuario;

    @ManyToMany(fetch = FetchType.LAZY)
    @JoinTable(name = "usuario_curso", joinColumns = @JoinColumn(name = "id_usuario"), inverseJoinColumns = @JoinColumn(name = "id_curso"))
    @JsonIgnoreProperties({ "clases_curso", "planes_curso", "precio_curso" })
    private List<Curso> cursos_usuario;

    @Temporal(TemporalType.DATE)
    private Date fecha_registro_usuario;

    @Column(name = "is_enabled")
    private Boolean isEnabled;

    @Column(name = "account_no_expired")
    private Boolean accountNoExpired;

    @Column(name = "account_no_locked")
    private Boolean accountNoLocked;

    @Column(name = "credentials_no_expired")
    private Boolean credentialsNoExpired;
}